File: python/api-tutorial-with-header.py
import requests

def getUrlToJSON(url, headers, payload):
    return requests.get(url, headers=headers, params=payload).json()
    
def generateURL(endpoint, options):
    params = []
    for key in options:
        params.append("{}={}".format(key, options[key]))
    return endpoint + "?" + "&".join(params)
    
apiKey = "SET-YOUR-API-KEY-HERE"

url = "https://min-api.cryptocompare.com/data/price"

payload = {
    "fsym": "BTC",
    "tsyms": "USD"
}

headers = {
    "authorization": apiKey
}

result = getUrlToJSON(url, headers, payload)
print(result)
"""
Result: 
{u'USD': 4069.35}
"""

File: node/api-tutorial-with-header.js
const request = require('request');

const apiKey = "SET-YOUR-API-KEY-HERE";

function getFullURL(url, options){
    const params = [];
    for (let key in options) {
        params.push(`${key}=${options[key]}`);
    }
    return url + '?' + params.join("&");
}

const baseURL = "https://min-api.cryptocompare.com/data/price";

const options = {
    fsym: "BTC",
    tsyms: "USD"
};

const headers = {
   "Authorization": apiKey 
};

const fullURL = getFullURL(baseURL, options);

request.get({
    url: fullURL,
    headers: headers
}, function(err, res, body){
    if (err){
        console.log(err);
    } else {
        console.log(body);
    }
});
